server/agents/project_chat_orchestrator.py

```diff
@@ -380,67 +390,71 @@ def _get_history(project_config: LlamaFarmConfig) -> ChatHistory:
-            return instructor.from_openai(openaiClient, mode=mode)
-        else:
-            return openaiClient
+    DEPRECATED: Use _get_client_for_model instead for multi-model support.
+    """
+    provider = get_provider(project_config.runtime.provider)
+    return provider.get_client(project_config)
 
-    else:
-        raise ValueError(f"Unsupported provider: {project_config.runtime.provider}")
 
+def _get_client_for_model(model_config):
+    """Get client for a specific model configuration using provider registry.
 
-def _determine_instructor_mode(project_config: LlamaFarmConfig) -> instructor.Mode:
-    # Use the configured instructor mode or default based on provider
-    if project_config.runtime.instructor_mode is not None:
-        # It's a string value
-        mode_str = project_config.runtime.instructor_mode
+    Args:
+        model_config: ModelConfig instance from ModelService
 
-        # Map the configured mode string to instructor.Mode
-        try:
-            mode = instructor.mode.Mode[mode_str.upper()]
-            logger.debug(f"Using configured instructor mode: {mode}")
-        except KeyError as e:
-            # Invalid mode specified
-            raise ValueError(
-                f"Invalid instructor_mode '{mode_str}'. "
-                f"Common modes include: tools, json, md_json, anthropic_tools, gemini_json. "
-                f"See instructor documentation for full list of supported modes."
-            ) from e
-    elif project_config.runtime.provider == Provider.ollama:
-        # Default to MD_JSON for Ollama as it's most compatible
-        mode = instructor.Mode.MD_JSON
-        logger.debug("Using MD_JSON mode for Ollama provider (default)")
-    else:
-        mode = instructor.Mode.TOOLS
-        logger.debug("Using TOOLS mode (default for non-Ollama)")
+    Returns:
+        AsyncOpenAI client (possibly instructor-wrapped)
+    """
+    from services.model_service import ModelConfig
+
+    provider = get_provider(model_config.provider)
+
+    # Create a temporary config-like object for the provider
+    # Providers expect LlamaFarmConfig but we only have model config
+    class TempConfig:
+        class Runtime:
+            def __init__(self, mc: ModelConfig):
+                self.provider = mc.provider
+                self.model = mc.model
+                self.base_url = mc.base_url
+                self.api_key = mc.api_key
+                self.huggingface_token = mc.huggingface_token
+                self.instructor_mode = mc.instructor_mode
+                self.prompt_format = mc.prompt_format
+                self.model_api_parameters = mc.model_api_parameters
+                self.lemonade = mc.lemonade
 
-    logger.debug(f"Instructor mode: {mode}")
-    return mode
+        def __init__(self, mc: ModelConfig):
+            self.runtime = self.Runtime(mc)
+
+    temp_config = TempConfig(model_config)
+    return provider.get_client(temp_config)  # type: ignore
```

Comment:
Do not add a type: ignore here. WHen you re-do this function to accept a model config, this type issue will be resolved.